runtime/common/src/asset_registry.rs

```diff
@@ -4,21 +4,37 @@ use frame_system::EnsureRoot;
 use orml_traits::asset_registry::{AssetMetadata, AssetProcessor};
 use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
 use scale_info::TypeInfo;
-use sp_runtime::DispatchError;
+use sp_core::Get;
+use sp_runtime::{BoundedVec, DispatchError};
+use sp_std::fmt::Debug;
 use spacewalk_primitives::CurrencyId;
 
-pub use spacewalk_primitives::CustomMetadata;
+#[derive(Clone, PartialOrd, Ord, PartialEq, Eq, Debug, Encode, Decode, TypeInfo, MaxEncodedLen)]
+pub struct CustomMetadata<T: Get<u32> + TypeInfo + Clone + Eq + Debug + Send + Sync> {
```

Comment:
All seems good to me!. Just wondering on this line, why does `T` has to implement `TypeInfo`, `Clone`, etc?
I think this is related to what @ebma said [here](https://github.com/pendulum-chain/pendulum/pull/398/files#r1456080048), which is going to be required if we have these bounds. Is it because we are passing `T` to `DiaKeys<T>`?

Reply:
It is needed for the implementation of `AssetProcessor`

Reply:
Oh I see now. AssetProcessor also has those bounds.